From 6e777f6691772853bd26f432f0a1ec17f28f746a Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Thu, 15 Nov 2018 18:30:49 -0700
Subject: [PATCH 4/5] common: Label and allow init to write to I/O sched tuning
 nodes

* LineageOS/android_system_sepolicy@0e3235f45d removed open, read,
  and write for init to general sysfs types. Our init scripts do
  fancy things with these nodes, so a specific type is necessary
  to allow such fancy things to happen in Pie+.

Change-Id: I15d8913ddf9758403e01f85c89ee326626094401

# Conflicts:
#	common/private/init.te
---
 common/private/file.te       | 1 +
 common/private/file_contexts | 3 +++
 common/private/init.te       | 2 ++
 3 files changed, 6 insertions(+)
 create mode 100644 common/private/file.te

diff --git a/common/private/file.te b/common/private/file.te
new file mode 100644
index 0000000..c039f4a
--- /dev/null
+++ b/common/private/file.te
@@ -0,0 +1 @@
+type sysfs_io_sched_tuneable, fs_type, sysfs_type;
diff --git a/common/private/file_contexts b/common/private/file_contexts
index 2c1a42c..d48fa49 100644
--- a/common/private/file_contexts
+++ b/common/private/file_contexts
@@ -5,6 +5,9 @@
 /system/bin/mkfs\.f2fs                  u:object_r:mkfs_exec:s0
 /system/bin/mkfs\.ntfs                  u:object_r:mkfs_exec:s0
 
+# I/O Scheduler
+/sys/devices/.*/block/(dm-|mmcblk|sd)[a-z0-9]+/queue(/.*)? u:object_r:sysfs_io_sched_tuneable:s0
+
 # OTA packages
 /data/lineageos_updates(/.*)?           u:object_r:ota_package_file:s0
 
diff --git a/common/private/init.te b/common/private/init.te
index bf6962d..edfaae7 100644
--- a/common/private/init.te
+++ b/common/private/init.te
@@ -1,2 +1,4 @@
 allow init sysfs:file { w_file_perms setattr };
 allow init sysfs_dm:file { w_file_perms setattr };
+
+allow init sysfs_io_sched_tuneable:file { setattr w_file_perms };
-- 
2.17.1

